bokehjs/src/lib/core/has_props.ts

```diff
@@ -314,7 +314,7 @@ export abstract class HasProps extends Signalable() implements Equatable, Printa
     // when loading a bunch of models, we want to do initialization as a second pass
     // because other objects that this one depends on might not be loaded yet
     if (deferred) {
-      assert(keys(attrs).length == 1)
+      assert(keys(attrs).length == 1, "'id' cannot be used together with property initializers")
```

Comment:
wouldn't asserting that `keys(attrs) == ["id"]` be more precise here? (and handle a case where there is one attr, and it is erroneously not `"id"`)

Reply:
This branch is only accessible when `id` is a key (it's checked earlier in the constructor), so here I'm only checking if there are no extra items. And yes, what you propose would be more readable, but:
```js
> ["id"] == ["id"]
false
```